### ISSUE TITLE
Support for platform MBeanServer

### ISSUE THREAD

**Issue by cruftex** [jsr107/jsr107tck#83]
The spec says on page 129:

> Simple in-process implementations might simply use the platform MBeanServer, accessed using
> ManagementFactory.getPlatformMBeanServer(). The caching implementation must document
> how to resolve the MBeanServer that was used to store cache management information.

However, the spec indirectly enforces that a separate MBeanServer is registered, because it wants to retrieve it by the id stored in the property `org.jsr107.tck.management.agentId`.

The Oracle Coherence does register its beans always with the platform MBeanServer. They sneaked around the TCK shortcoming by providing a server wrapper which constantly produces the id `Coherence WrapperMBeanServer` for the platform MBeanServer.

All other existing implementations did copy the approach of the RI and provided a custom implementation   of MBeanServerBuilder to produce a separate MBeanServer which the TCK can pick up.

The TCK should support the simple approach, registering the bean with the platform MBeanServer, straight forward. All implementations except Oracle produced extra code to register a separate MBeanServer to comply with the TCK, however, the spec does not require that.

At other places the Spec clearly says, that implementation _must_ register with the platform mbean server. See JavaDoc of `CacheManager.enableManagement()`.

> Controls whether management is enabled. If enabled the {@link CacheMXBean}
> for each cache is registered in the platform MBean server. The platform
> MBeanServer is obtained using
> {@link ManagementFactory#getPlatformMBeanServer()}.

The Spec should be clear whether this is mandatory or not at all places.

**Comment by cruftex** [comment]
The needed change is trivial. The method `TestSupport.resolveMBeanServer()` is now:

```` java
  public static MBeanServer resolveMBeanServer() {

    if (System.getProperty("javax.management.builder.initial") == null) {
      throw new RuntimeException("You must set the 'javax.management.builder" +
          ".initial' property so that the TCK can find your MBeanServer'");
    }
    if (System.getProperty("org.jsr107.tck.management.agentId") == null) {
      throw new RuntimeException("You must set the 'org.jsr107.tck.management.agentId'" +
          "system property so that the TCK can find your MBeanServer agent by " +
          "its ID.'");
    }

    String agentId = System.getProperty("org.jsr107.tck.management.agentId");
    ArrayList<MBeanServer> mBeanServers = MBeanServerFactory.findMBeanServer(agentId);
    if (mBeanServers.size() < 1) {
      throw new CacheException("The specification requires registration of " +
          "MBeans in an implementation specific MBeanServer. A search for an " +
          "MBeanServer did not find any.");
    }
    return mBeanServers.get(0);
  }
````

It can be changed to:

```` java
  public static MBeanServer resolveMBeanServer() {
    String agentId = System.getProperty("org.jsr107.tck.management.agentId");
    ArrayList<MBeanServer> mBeanServers = MBeanServerFactory.findMBeanServer(agentId);
    if (mBeanServers.size() < 1) {
      throw new CacheException("The specification requires registration of " +
          "MBeans in an implementation specific MBeanServer. A search for an " +
          "MBeanServer did not find any.");
    }
    return mBeanServers.get(0);
  }
````

If `agentId` is `null`, just the first (and only) MBeanServer gets picked.

I can check this thoroughly this and make a pull request. Any comments?

**Comment by ljacomet** [comment]
@cruftex Ehcache 3 also takes the approach of registering MBeans with the platform `MBeanServer` and provides a bridge for the TCK execution.

I agree it would be nice if the TCK was testing the real runtime behaviour and not a fabricated one.

**Comment by cruftex** [comment]
Preparing an PR.

Cache implementations need to change behavior and register with the platform MBean server.

The TCK guide needs an update. The properties `javax.management.builder.initial` and `org.jsr107.tck.management.agentId` are no longer needed.

Cache implementations that want to make the transition before TCK 1.1 is released, can use some glue code that makes the platform MBean server available to the TCK under a specified ID. See, for example: https://github.com/cache2k/cache2k/blob/master/cache2k-jcache/src/main/java/org/cache2k/jcache/provider/tckGlue/TckMBeanServerBuilder.java

**Comment by cruftex** [comment]
PRs for RI and TCK are ready.

**Comment by vbekiaris** [comment]
Fixed by #119 & https://github.com/jsr107/RI/issues/56

**Comment by gregrluck** [comment]
The two properties were not removed from the surefire plugin.
